activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -87,7 +87,8 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
         old_name = old_name.to_s
 
         method_defined = method_defined?(target_name) || private_method_defined?(target_name)
-        manually_defined = method_defined && self.instance_method(target_name).owner != generated_attribute_methods
+        manually_defined = method_defined &&
+          !self.instance_method(target_name).owner.is_a?(generated_attribute_methods.class)
```

Comment:
```suggestion
          !self.instance_method(target_name).owner.is_a?(GeneratedAttributeMethods)
```

Might be clearer, no?

Reply:
Indeed it has a better readability 👍 Changed